conans/test/functional/package_id/package_id_requires_modes_test.py

```diff
@@ -455,3 +455,36 @@ def test_std_matching_with_compiler_cppstd(self):
                         ' -s compiler="gcc" -s compiler.libcxx=libstdc++11'
                         ' -s compiler.version=7.2 -s compiler.cppstd=gnu14')
         self.assertIn("Hello/1.2.0@user/testing: Already installed!", self.client.out)
+
+    def test_package_id_requires_patch_mode(self):
+        channel = "user/testing"
+        self.client.run("config set general.default_python_requires_id_mode=patch_mode")
+        self._export("liba", "0.1.0", channel=channel, package_id_text=None, requires=None)
+        self.client.run("create . liba/0.1.0@user/testing")
+        self._export("libb", "0.1.0", channel=channel, package_id_text=None, requires=["liba/0.1.0@user/testing"])
+        self._export("libc", "0.1.0", channel=channel, package_id_text=None, requires=["liba/0.1.0@user/testing", "libb/0.1.0@user/testing"])
+        self.assertIn("libc/0.1.0@user/testing: A new conanfile.py version was exported", self.client.out)
+        self.client.run("create . libc/0.1.0@user/testing", assert_error=True)
+
+        self.assertIn("""libc/0.1.0@user/testing: Forced build from source
+Installing package: libc/0.1.0@user/testing
+Requirements
+    liba/0.1.0@user/testing from local cache - Cache
+    libb/0.1.0@user/testing from local cache - Cache
+    libc/0.1.0@user/testing from local cache - Cache
+Packages
+    liba/0.1.0@user/testing:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Cache
+    libb/0.1.0@user/testing:072b0436dd87762380c57324afddfb7e7aa3e872 - Missing
+    libc/0.1.0@user/testing:20f4b9ad6adeec63c71573d5a6db78b03dbdf919 - Build
+
+Installing (downloading, building) binaries...
+ERROR: Missing binary: libb/0.1.0@user/testing:072b0436dd87762380c57324afddfb7e7aa3e872
+
+libb/0.1.0@user/testing: WARN: Can't find a 'libb/0.1.0@user/testing' package for the specified settings, options and dependencies:
+- Settings: 
+- Options: an_option=off, liba:an_option=off
+- Dependencies: liba/0.1.0@user/testing
+- Requirements: liba/0.1.0
```

Comment:
I think we need at least one more level in the graph (maybe we should add one header-only in the middle) and check that we see here all the requirements in the closure (all the packages that contribute to the package_id). 

Note.- Depending on the value of `full_transitive_package_id` (https://github.com/conan-io/conan/pull/6451) that list should be different (but that should be something to check associated with that feature, not here).

Reply:
All those virtual packages are header-only, because I removed all settings. Do you think is better adding libraries + header-only? We will have different package_id according the OS, it's hard to filter, but we can add for Linux, Windows and Mac at least.

Reply:
Here we are interested in the `requirements` list, but only if it adds valuable information to the user. I think that the idea motivating this PR is to list all the packages in the closure not just the direct requirements (maybe we are already doing it, I don't know).

The missing package-ID depends, not only on `liba/0.1.0@user/testing`, but also in the transitive requirements (exception #6451) and I think this is what we have to check here: that all the requirements, also the transitive ones, are listed here. We don't need to check the actual package_ID value.

Reply:
The ``ConanInfo.requirements`` field should contain all references that are involved in the computation of the packageID, including the transitive ones, and I think this is valuable information.

Reply:
now it makes sense!

Reply:
okay, now I think that test looks better. Tell me what you guys think.